### ISSUE TITLE
[2.8.0]:  Setting `addRepaintBoundaries` to `true` or `false` shows blinking horizontal line while scrolling.

### ISSUE THREAD

**Issue by Haleydu** [flutter/flutter#95199]
I upgraded flutter from 2.5.1 to 2.8.0 . Tests found the addRepaintBoundaries Invalidation of the ScrollablePositionedList. There is no effective  addRepaintBoundaries set true or false. But it is normal in flutter 2.5.1

**Comment by darshankawar** [comment]
@Haleydu 
Can you elaborate on what issue are you facing exactly using a code sample ?
Thanks.

**Comment by Haleydu** [comment]
When loading a series of images, (addRepaintBoundaries)whether set to true or false, They all have a redrawn black line
There are no black lines when flutter2.5.1～2.5.3 is set(addRepaintBoundaries) to false

**Comment by darshankawar** [comment]
@Haleydu 
Please provide us a [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) as requested earlier to properly address this.
Thanks.

**Comment by Haleydu** [comment]
```
final ItemScrollController itemScrollController = ItemScrollController();
  final ItemPositionsListener itemPositionsListener =
      ItemPositionsListener.create();
  var headers = {
  "user-agent":
  "Mozilla/5.0 (iPhone; CPU iPhone OS 12_0 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/12.0 Mobile/15A372 Safari/604.1"
  };
  var listImageUrl = [
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/1.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/2.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/3.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/4.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/5.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/6.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/7.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/8.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/9.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/10.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/11.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/12.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/19-ycot/1.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/19-ycot/2.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/18.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/1.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/2.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/3.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/4.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/5.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/6.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/7.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/8.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/9.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/10.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/11.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/12.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/13.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/14.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/15.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/16.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/17.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/18.jpg"
  ];
  Widget createImageList() {
    return InkWell(
      onTap: () {},
      child: Stack(
        children: [
          ScrollablePositionedList.builder(
            ///flutter2.8.0（invalid）, try set true or false
            ///flutter2.5.1（normal）, set true or false
            addRepaintBoundaries: false,
            initialScrollIndex: 1,
            scrollDirection: Axis.vertical,
            itemScrollController: itemScrollController,
            itemPositionsListener: itemPositionsListener,
            itemCount: listImageUrl.length,
            itemBuilder: (BuildContext context, int index) {
              return LayoutBuilder(
                builder: (BuildContext context, BoxConstraints constraints) {
                  Size renderSize =
                      Size(constraints.maxWidth, constraints.maxWidth / 2);
                  return CachedNetworkImage(
                    imageUrl: listImageUrl[index],
                    httpHeaders: headers,
                    filterQuality: FilterQuality.high,
                    progressIndicatorBuilder:
                        (context, url, downloadProgress) => SizedBox(
                      width: renderSize.width,
                      height: renderSize.height,
                      child: Center(
                        child: CircularProgressIndicator(
                            value: downloadProgress.progress),
                      ),
                    ),
                  );
                },
              );
            },
          ),
        ],
      ),
    );
  }
```

**Comment by darshankawar** [comment]
@Haleydu 
Still doesn't seem to be complete code that I can directly run. Where are `ScrollablePositionedList`, `ItemScrollController`
and `ItemPositionsListener` coming from ?

**Comment by Haleydu** [comment]
You can also replace  **ScrollablePositionedList** with **ListView**

**Comment by Haleydu** [comment]
```dart
import 'package:cached_network_image/cached_network_image.dart';
import 'package:flutter/material.dart';

class ImageListScreen extends StatefulWidget {

  const ImageListScreen({Key? key}) : super(key: key);

  @override
  _ImageListScreenState createState() => _ImageListScreenState();
}

class _ImageListScreenState extends State<ImageListScreen> {
  @override
  Widget build(BuildContext context) {
    return buildScaffold(context);
  }

  Widget buildScaffold(BuildContext context) {
    return Scaffold(
        /// The key is to set this to black，
        /// If you set it to green or blue it will have a corresponding green line or blue line when you slide it
        backgroundColor: Colors.black,
        body: createImageList());
  }

  var headers = {
    "user-agent":
        "Mozilla/5.0 (iPhone; CPU iPhone OS 12_0 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/12.0 Mobile/15A372 Safari/604.1"
  };
  var listImageUrl = [
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/1.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/2.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/3.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/4.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/5.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/6.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/7.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/8.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/9.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/10.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/11.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/17-6hku/12.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/19-ycot/1.jpg",
    "https://s2.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/19-ycot/2.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/18.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/1.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/2.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/3.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/4.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/5.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/6.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/7.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/8.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/9.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/10.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/11.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/12.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/13.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/14.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/15.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/16.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/17.jpg",
    "https://s1.baozimh.com/fcomic/zhilengshaonu-zhilengshaonu/0/10-fqt8/18.jpg"
  ];

  Widget createImageList() {
    return InkWell(
      onTap: () {},
      child: Stack(
        children: [
          ListView.builder(
            ///flutter2.8.0（invalid）, try set addRepaintBoundaries true or false
            ///flutter2.5.1（normal）, set addRepaintBoundaries true or false
            addRepaintBoundaries: false,
            scrollDirection: Axis.vertical,
            itemCount: listImageUrl.length,
            itemBuilder: (BuildContext context, int index) {
              return LayoutBuilder(
                builder: (BuildContext context, BoxConstraints constraints) {
                  Size renderSize =
                      Size(constraints.maxWidth, constraints.maxWidth / 2);
                  return CachedNetworkImage(
                    imageUrl: listImageUrl[index],
                    httpHeaders: headers,
                    filterQuality: FilterQuality.high,
                    progressIndicatorBuilder:
                        (context, url, downloadProgress) => SizedBox(
                      width: renderSize.width,
                      height: renderSize.height,
                      child: Center(
                        child: CircularProgressIndicator(
                            value: downloadProgress.progress),
                      ),
                    ),
                  );
                },
              );
            },
          ),
        ],
      ),
    );
  }
}

```

**Comment by Haleydu** [comment]
Uploading Screenrecorder-2021-12-16-16-11-24-222.mp4…

**Comment by Haleydu** [comment]
Uploading Screenrecorder-2021-12-16-16-11-24-222.mp4…

**Comment by Haleydu** [comment]
https://user-images.githubusercontent.com/44172947/146334003-174a435e-c871-4917-b5df-9a8f567c70f4.mp4

**Comment by darshankawar** [comment]
Thanks for the update @Haleydu. The blinking horizontal line while scrolling is the issue you are trying to report ?

**Comment by Haleydu** [comment]
yes， The blinking horizontal line did not happen with flutter 2.5.1

**Comment by darshankawar** [comment]
Thanks. I see the same behavior (ie, blinking horizontal line while scrolling up and down) on latest stable (2.8.1) and on latest master(2.9.0-1.0.pre.122):


https://user-images.githubusercontent.com/67046386/146495731-95d353cf-25fa-4513-9e84-d132488a2dde.mov


Although the same functionality worked properly on stable 2.5.3:


https://user-images.githubusercontent.com/67046386/146495824-933eb0d6-a302-42dd-a851-b4c965012439.mov


<details>
<summary> stable, master flutter doctor -v </summary>

```
[[✓] Flutter (Channel stable, 2.8.1, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.8.1 at /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (13 hours ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/dhs/Library/Android/sdk
    • Java binary at: /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.    

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 93.0.4577.82


• No issues found!

[✓] Flutter (Channel master, 2.9.0-1.0.pre.122, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.9.0-1.0.pre.122 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c8981c5fbd (3 hours ago), 2021-12-16 21:19:08 -0500
    • Engine revision d8c82c5469
    • Dart version 2.16.0 (build 2.16.0-116.0.dev)
    • DevTools version 2.9.1
    
[✓] Android toolchain - develop for Android devices (Android SDK version 30)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/dhs/Library/Android/sdk
    • Java binary at: /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.    

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 93.0.4577.82


• No issues found!




```

</details>

**Comment by coldwaterfish** [comment]
flutter 2.8.1
android
I see the blinking horizontal line，I don't have the reason

**Comment by HansMuller** [comment]
This looks like a known problem with antialiasing and mapping from logical pixels to device pixels. There's an open issue for a documentation fix: #14288

More info about the problem can be found here:
https://github.com/flutter/flutter/issues/14288#issuecomment-361375791
https://github.com/flutter/flutter/issues/17084#issuecomment-385718108
https://github.com/flutter/flutter/issues/15035#issuecomment-370028740

**Comment by Piinks** [comment]
This does look like an antialiasing issue as @HansMuller mentioned above. I am going to link this to the canonical issue (#14288) for antialiasing and close it. Setting your background color to be the same as the background of your scroll view may prevent these from being visible.
